packages/react-provider/src/hooks/use-client-source-id.ts

```diff
@@ -12,9 +9,22 @@ const useClientSourceId = ({
   id,
   localStorage,
 }): string => {
-  const clientSourceKey = useHash(
-    authorization ? authorization : `${clientKey}/${userId}`
-  );
+  const clientSourceKey = useMemo(() => {
+    if (!authorization) {
+      return `${clientKey}/${userId}`;
+    }
+
+    const decoded = jwtDecode(authorization) as {
+      tenantId: string;
+      scope: string;
+    };
+    const scopeUserId = decoded?.scope
```

Comment:
not sure how this would work with multiple userids

Reply:
but we don't have anyone doing that yet...  i think it opens a can of works to conceptualize